新增本地文件数据库存储方式

myadream opened the pull request:
* [x] 新增单词
* [x] 单词列表
* [x] 单词搜索、标签搜索、状态搜索
* [x] 单词复习
* [x] 单词学习记录面板 
* [x] 单词数据库新增编辑、刷新、新增按钮 
* [x] 新增通过列句复习方式
* [x] 调整单词添加、编辑通过弹框形式处理
* [ ] 导入、导出

myadream committed:
update

myadream committed:
调试

myadream committed:
调整底层使用的数据库

myadream committed:
调整底层使用的数据库

myadream committed:
update

myadream committed:
新增本地文件数据库

myadream committed:
update

myadream committed:
udpate

myadream committed:
udpate

myadream committed:
update

myadream committed:
update

myadream committed:
新增本地文件存储类型

myadream committed:
update

myadream commented:
![image](https://github.com/guopenghui/obsidian-language-learner/assets/40756102/dbcff4c7-ff19-4234-99fb-f997bb62049f)

myadream committed:
修复部分问题

myadream committed:
修复刷新复习数据库状态筛选问题

myadream committed:
修复学习单词统计面板筛选问题

wuchaooooo commented:
可以尽快把这个功能合并到master，打包成release使用吗～ @guopenghui

e-zz commented:
> 可以尽快把这个功能合并到master，打包成release使用吗～ @guopenghui

https://github.com/e-zz/obsidian-language-learner/releases/tag/db-release
目前插件的原作者似乎不再积极维护了。所以这种重要的功能更新最好谨慎使用，试过再正式升级到以往的 vault，以免遇到恶性 bug

gaoluji commented:
> > 可以尽快把这个功能合并到master，打包成release使用吗～ @guopenghui
> 
> https://githubfast.com/e-zz/obsidian-language-learner/releases/tag/db-release 目前插件的原作者似乎不再积极维护了。所以这种重要的功能更新最好谨慎使用，试过再正式升级到以往的 vault，以免万一遇到恶性 bug

例句和笔记无法加载
![image](https://github.com/guopenghui/obsidian-language-learner/assets/139356383/276021eb-7420-468e-a24f-014b3e72c18f)
![image](https://github.com/guopenghui/obsidian-language-learner/assets/139356383/868b1723-069d-49d3-ac66-cee6b78def25)
例句之类的都没法导入间隔重复界面

myadream commented:
> > > 可以尽快把这个功能合并到master，打包成release使用吗～ @guopenghui
> > 
> > 
> > https://githubfast.com/e-zz/obsidian-language-learner/releases/tag/db-release 目前插件的原作者似乎不再积极维护了。所以这种重要的功能更新最好谨慎使用，试过再正式升级到以往的 vault，以免万一遇到恶性 bug
> 
> 例句和笔记无法加载 ![image](https://user-images.githubusercontent.com/139356383/274153929-276021eb-7420-468e-a24f-014b3e72c18f.png) ![image](https://user-images.githubusercontent.com/139356383/274154061-868b1723-069d-49d3-ac66-cee6b78def25.png) 例句之类的都没法导入间隔重复界面

感谢~ 这几天我抽空排除一下问题

gaoluji commented:
> > > > 可以尽快把这个功能合并到master，打包成release使用吗～ @guopenghui
> > > 
> > > 
> > > https://githubfast.com/e-zz/obsidian-language-learner/releases/tag/db-release 目前插件的原作者似乎不再积极维护了。所以这种重要的功能更新最好谨慎使用，试过再正式升级到以往的 vault，以免万一遇到恶性 bug
> > 
> > 
> > 例句和笔记无法加载 ![image](https://usercontent.githubfast.com/user-images/139356383/274153929-276021eb-7420-468e-a24f-014b3e72c18f.png) ![image](https://usercontent.githubfast.com/user-images/139356383/274154061-868b1723-069d-49d3-ac66-cee6b78def25.png) 例句之类的都没法导入间隔重复界面
> 
> 感谢~ 这几天我抽空排除一下问题

https://github.com/oflg/Tidme/issues/8
一种新的背单词方式

gaoluji commented:
> > > > 可以尽快把这个功能合并到master，打包成release使用吗～ @guopenghui
> > > 
> > > 
> > > https://githubfast.com/e-zz/obsidian-language-learner/releases/tag/db-release 目前插件的原作者似乎不再积极维护了。所以这种重要的功能更新最好谨慎使用，试过再正式升级到以往的 vault，以免万一遇到恶性 bug
> > 
> > 
> > 例句和笔记无法加载 ![image](https://usercontent.githubfast.com/user-images/139356383/274153929-276021eb-7420-468e-a24f-014b3e72c18f.png) ![image](https://usercontent.githubfast.com/user-images/139356383/274154061-868b1723-069d-49d3-ac66-cee6b78def25.png) 例句之类的都没法导入间隔重复界面
> 
> 感谢~ 这几天我抽空排除一下问题

大佬，复习样式能不能改成通过例句来猜单词意思

gaoluji commented:
> > > > 可以尽快把这个功能合并到master，打包成release使用吗～ @guopenghui
> > > 
> > > 
> > > https://githubfast.com/e-zz/obsidian-language-learner/releases/tag/db-release 目前插件的原作者似乎不再积极维护了。所以这种重要的功能更新最好谨慎使用，试过再正式升级到以往的 vault，以免万一遇到恶性 bug
> > 
> > 
> > 例句和笔记无法加载 ![image](https://usercontent.githubfast.com/user-images/139356383/274153929-276021eb-7420-468e-a24f-014b3e72c18f.png) ![image](https://usercontent.githubfast.com/user-images/139356383/274154061-868b1723-069d-49d3-ac66-cee6b78def25.png) 例句之类的都没法导入间隔重复界面
> 
> 感谢~ 这几天我抽空排除一下问题

我看了一下，只要把闪卡文档里面的复习样式给改了就能实现

myadream commented:
> > > > > 可以尽快把这个功能合并到master，打包成release使用吗～ @guopenghui
> > > > 
> > > > 
> > > > https://githubfast.com/e-zz/obsidian-language-learner/releases/tag/db-release 目前插件的原作者似乎不再积极维护了。所以这种重要的功能更新最好谨慎使用，试过再正式升级到以往的 vault，以免万一遇到恶性 bug
> > > 
> > > 
> > > 例句和笔记无法加载 ![image](https://usercontent.githubfast.com/user-images/139356383/274153929-276021eb-7420-468e-a24f-014b3e72c18f.png) ![image](https://usercontent.githubfast.com/user-images/139356383/274154061-868b1723-069d-49d3-ac66-cee6b78def25.png) 例句之类的都没法导入间隔重复界面
> > 
> > 
> > 感谢~ 这几天我抽空排除一下问题
> 
> 我看了一下，只要把闪卡文档里面的复习样式给改了就能实现

不是很确定...  因为我还没有进行看是怎么实现的..

gaoluji commented:
> > > > > > 可以尽快把这个功能合并到master，打包成release使用吗～ @guopenghui
> > > > > 
> > > > > 
> > > > > https://githubfast.com/e-zz/obsidian-language-learner/releases/tag/db-release 目前插件的原作者似乎不再积极维护了。所以这种重要的功能更新最好谨慎使用，试过再正式升级到以往的 vault，以免万一遇到恶性 bug
> > > > 
> > > > 
> > > > 例句和笔记无法加载 ![image](https://usercontent.githubfast.com/user-images/139356383/274153929-276021eb-7420-468e-a24f-014b3e72c18f.png) ![image](https://usercontent.githubfast.com/user-images/139356383/274154061-868b1723-069d-49d3-ac66-cee6b78def25.png) 例句之类的都没法导入间隔重复界面
> > > 
> > > 
> > > 感谢~ 这几天我抽空排除一下问题
> > 
> > 
> > 我看了一下，只要把闪卡文档里面的复习样式给改了就能实现
> 
> 不是很确定... 因为我还没有进行看是怎么实现的..

![image](https://github.com/guopenghui/obsidian-language-learner/assets/139356383/a912c843-56ac-4c5d-81ea-e7fc9f7ccbf5)
目前是这样保存单词的，实际上还是过去背单词的方式：单词》意思
可以改成这样，通过句子检测单词意思。
![image](https://github.com/guopenghui/obsidian-language-learner/assets/139356383/1035c2ef-be98-436f-bf87-e71afbbbe0d9)
句子都是阅读过程中得到的。
![image](https://github.com/guopenghui/obsidian-language-learner/assets/139356383/d55d1084-4a36-4de6-8531-1ad707f9f6f3)
![image](https://github.com/guopenghui/obsidian-language-learner/assets/139356383/fde6e807-823f-4141-a104-a1770ddea41e)
这个实际上更加符合二语习得

gaoluji commented:
而且插件本身可以管理句子
![image](https://github.com/guopenghui/obsidian-language-learner/assets/139356383/c64eb1a4-5906-4e4b-8b8c-35ff973eb70a)
![image](https://github.com/guopenghui/obsidian-language-learner/assets/139356383/e8777cda-e85e-45ce-ac6d-366b7fb8ae47)
笔记功能可以用于对每一个例句进行想写的笔记
![image](https://github.com/guopenghui/obsidian-language-learner/assets/139356383/676e0bcd-9b29-4700-8481-f54b1dd6256d)
含义应当是单词在每个例句下的意义（因为在我们实际的英语使用中，我们都是在一个句子里，并且在句子里单词的意义通常是单一确定的），而目前的单词实际上还是普通背单词软件的样式，单词》多个意义。
ps：能否开一个新的issue，这里太不方便提意见了（
@myadream

myadream committed:
update

myadream committed:
update

myadream committed:
学习单词页面调整

1.  单词库新增编辑功能和新增单词按钮
2. 修复闪卡笔记记录问题

myadream committed:
修复单词不能重复编辑问题

myadream committed:
update

myadream committed:
update

myadream committed:
update manifest

myadream committed:
update manifest

myadream committed:
update

myadream committed:
update

myadream committed:
解决冲突

myadream committed:
update

SlionRoy commented:
您好，有考虑移动端适配吗

myadream commented:
> 您好，有考虑移动端适配吗

好像没有区分吧。我抽空试试看有没有问题

myadream committed:
新增 sqlite3 存储方式

myadream committed:
优化 DataPanel.vue 显示和筛选逻辑

myadream committed:
优化 DataPanel.vue 显示和筛选逻辑

myadream committed:
优化 DataPanel.vue 显示和筛选逻辑

myadream committed:
优化 DataPanel.vue 显示和筛选逻辑

myadream committed:
优化 DataPanel.vue 显示和筛选逻辑

myadream committed:
优化 DataPanel.vue 显示和筛选逻辑

myadream committed:
优化 DataPanel.vue 显示和筛选逻辑

myadream committed:
打包工具迁移至 vite , 优化 sqlite 存储和DataPanel 页面显示

myadream committed:
打包工具迁移至 vite , 优化 sqlite 存储和DataPanel 页面显示

myadream committed:
打包工具迁移至 vite , 优化 sqlite 存储和DataPanel 页面显示

myadream committed:
build(release): 添加 sql-wasm.wasm 到发布流程

- 在构建步骤中包含 sql-wasm.wasm 文件到插件目录
- 添加 sql-wasm.wasm 作为发布资产上传
- 配置 WASM 文件的正确 MIME 类型为 application/wasm
- 确保发布包中包含所有必要的 WASM 依赖文件

myadream committed:
update manifest

myadream committed:
update manifest

myadream committed:
调整workflows node 版本

myadream committed:
update manifest

myadream committed:
修复搜索单词时添加问题

myadream committed:
update manifest

myadream committed:
支持移动端方案

myadream committed:
统一移动端、pc 端显示方式 DataPanel.vue

myadream committed:
统一移动端、pc 端显示方式 DataPanel.vue

myadream committed:
update manifest

myadream committed:
update

myadream committed:
update

myadream committed:
update

myadream committed:
调整移动端兼容性

myadream committed:
update manifest

myadream committed:
update manifest

myadream committed:
兼容文件同步后的数据库未重新打开读取数据问题

myadream committed:
DataPanel.vue 优化数据加载方式

myadream committed:
update manifest
